### PR TITLE
Added isDirectory property to pathProperties

### LINKED ISSUE #9525: Add an isDirectory property to PathProperties
https://github.com/Azure/azure-sdk-for-net/issues/10796

### PR DESCRIPTION
**Pull request by gapra-msft**
Resolves #9525

### CODE REVIEW

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/PathProperties.java [hunk 1]
```diff
@@ -107,116 +110,123 @@ public PathProperties(final OffsetDateTime creationTime, final OffsetDateTime la
         this.encryptionKeySha256 = encryptionKeySha256;
         this.accessTierChangeTime = accessTierChangeTime;
         this.metadata = metadata;
+        /* Default isDirectory to false. */
+        if (this.metadata == null) {
+            this.isDirectory = false;
+        } else {
+            this.isDirectory = String.valueOf(true)
+                .equalsIgnoreCase(metadata.get(Constants.HeaderConstants.DIRECTORY_METADATA_KEY));
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/PathProperties.java]: Mind replacing this with `Boolean.parseBoolean(metadata.get(Constants.HeaderConstants.DIRECTORY_METADATA_KEY))`, it does the same thing.

File: sdk/storage/azure-storage-file-datalake/CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,8 @@
 # Release History
 
 ## 12.1.0-beta.1 (Unreleased)
-
+- Fixed a NPE caused due to deserializing a non existant lastModifiedTime.
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-file-datalake/CHANGELOG.md]: Was this actually done in this PR? I'm not really seeing a change that would indicate this.

Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/CHANGELOG.md]: No, I did it in a PR earlier this week, just forgot to update Changelog

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/PathProperties.java (hunk 1 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/PathProperties.java]: done

**Pull request merged**